inference/providers/leptonai/chat_completion.py

```diff
@@ -0,0 +1,212 @@
+    if message.role in [
+        ChatCompletionRole.user,
+        ChatCompletionRole.system,
+    ] or is_assistant_text_message(message):
+        return {"role": message.role.name, "content": message.content}
+
+    if message.role == ChatCompletionRole.function:
+        message: ChatCompletionFunctionMessage
+        return {"role": "tool", "content": message.content, "tool_call_id": message.id}
+
+    if is_assistant_function_calls_message(message):
+        message: ChatCompletionAssistantMessage
+        function_calls = []
+
+        for f in message.function_calls:
+            arguments = f.arguments
+            if isinstance(arguments, dict):
+                arguments = json.dumps(arguments)
+            function_calls.append(
+                {
+                    "id": f.id,
+                    "type": "function",
+                    "function": {"name": f.name, "arguments": arguments},
+                }
+            )
+
+        return {
+            "role": ChatCompletionRole.assistant.name,
+            "tool_calls": function_calls,
+            "content": None,
+        }
+
+
+def _build_leptonai_chat_completion_payload(
+    messages: List[ChatCompletionMessage],
+    stream: bool,
+    provider_model_id: str,
+    configs: ChatCompletionModelConfiguration,
+    function_call: Optional[str],
+    functions: Optional[List[ChatCompletionFunction]],
+):
+    # Convert ChatCompletionMessages to the required format
+    formatted_messages = [_build_leptonai_message(msg) for msg in messages]
+    logger.debug("formatted_messages: %s", formatted_messages)
+    payload = {
+        "messages": formatted_messages,
+        "model": provider_model_id,
+        "stream": stream,
+    }
+    config_dict = configs.model_dump()
+    for key, value in config_dict.items():
+        if value is not None:
+            payload[key] = value
+    if function_call:
+        if function_call in ["none", "auto"]:
+            payload["tool_choice"] = function_call
+        else:
+            payload["tool_choice"] = {"name": function_call}
+    if functions:
+        payload["tools"] = [{"type": "function", "function": f.model_dump()} for f in functions]
+    return payload
+
+
+class LeptonaiChatCompletionModel(BaseChatCompletionModel):
+    def __init__(self):
+        super().__init__()
+
+    # ------------------- prepare request data -------------------
+
+    def prepare_request(
+        self,
+        stream: bool,
+        provider_model_id: str,
+        messages: List[ChatCompletionMessage],
+        credentials: ProviderCredentials,
+        configs: ChatCompletionModelConfiguration,
+        function_call: Optional[str] = None,
+        functions: Optional[List[ChatCompletionFunction]] = None,
+    ) -> Tuple[str, Dict, Dict]:
+        # todo accept user's api_url, the resulting URL will be a specific endpoint for a Lepton AI chat model
+        if provider_model_id not in model_id_list:
```

Comment:
Directly pass provider_model_id to the provider. Raise provider error if the model_id not found